net-libs/librsync: fix testsuite for v0.9.7, bug #511954

axelgenus opened the pull request:
I added a patch (from Ubuntu lauchpad) to fix compilation with FEATURES="test".

idella commented:
The commit msg need reflect or report the fix.  fix #511954 #906 are merely tags or refs without reporting what was done. More importantly this does not qualify for a revbump. It is fixing a test which is run within the ${S} and make no impact on the state of the files finally installed. Test phases are run then all traces deleted on install. Can you re-edit to similar to:

net-libs/librsync: fix to testsuite in vn. 0.9.7-r3

Add patch librsync-0.9.7-fix-tests.patch to effect patch,
patch sourced from ......

Gentoo bug: #.511954


Also, Author: Andrey Rahmatullin <wrar@wrar.name> tells us you have acquired this patch form this author. Ideally, the first line of the patch itself ought supply a link to the a commit to an online repo.

gktrk commented:
The Manifest file looks a bit suspicious. We have [thin manifests](https://wiki.gentoo.org/wiki/Repository_format/package/Manifest#Thin_Manifest) enabled for the Gentoo repo which means '.ebuild', 'metadata.xml', and '.patch' files are not part of the Manifest now. Using 'repoman commit' should help you avoid these problems as it regenerates the manifest before the commit.

axelgenus commented:
@idella you're correct: the patch only fixes test suite so no need for a revbump. I reverted that in the new commit. As for the patch, I took it from the .deb file in the Debian repository[1]. The patch is from Andrey Rahmatullin[2] which is the librsync1 maintainer for Debian. Debian bug #713443 seems to be the same as Gentoo bug #511954 but the patch has not been attached to the report. It seems that it has been directly committed to Debian repository.

@gktrk I've fixed the Manifest but I don't know what happened. I used "ebuild <file> manifest" and it added all files to the Manifest.

[1] https://packages.debian.org/jessie/librsync1
[2] https://github.com/wRAR
